tests/helpers/test_service.py

```diff
@@ -307,20 +325,50 @@ async def test_async_get_all_descriptions(hass):
 
 
 async def test_call_with_required_features(hass, mock_entities):
-    """Test service calls invoked only if entity has required feautres."""
+    """Test service calls invoked only if entity has required features."""
     test_service_mock = AsyncMock(return_value=None)
     await service.entity_service_call(
         hass,
         [Mock(entities=mock_entities)],
         test_service_mock,
         ha.ServiceCall("test_domain", "test_service", {"entity_id": "all"}),
-        required_features=[1],
+        required_features=[SUPPORT_A],
     )
-    assert len(mock_entities) == 2
+    assert len(mock_entities) == 4
+    # Called once because only one of the entities had the required features
+    assert test_service_mock.call_count == 2
+
+
+async def test_call_with_both_required_features(hass, mock_entities):
+    """Test service calls invoked only if entity has required features."""
+    test_service_mock = AsyncMock(return_value=None)
+    await service.entity_service_call(
+        hass,
+        [Mock(entities=mock_entities)],
+        test_service_mock,
+        ha.ServiceCall("test_domain", "test_service", {"entity_id": "all"}),
+        required_features=[SUPPORT_A | SUPPORT_B],
+    )
+    assert len(mock_entities) == 4
```

Comment:
It will always be four entities and those are part of our setup so we don't need to assert it.